ci-operator/step-registry/azure/provision/service-principal/minimal-permission/azure-provision-service-principal-minimal-permission-commands.sh

```diff
@@ -136,282 +104,42 @@ fi
-# for destroy
-echo "${sp_id}" > "${SHARED_DIR}/azure_sp_id"
-rm -f ${SP_OUTPUT}
-
-# create SP with minimal permission for CCO to create required Azure resources when using workload identity
-if [[ "${ENABLE_MIN_PERMISSION_FOR_STS}" == "true" ]]; then
-    sts_required_permissions="""
-\"Microsoft.Resources/subscriptions/resourceGroups/read\",
-\"Microsoft.Resources/subscriptions/resourceGroups/write\",
-\"Microsoft.Resources/subscriptions/resourceGroups/delete\",
-\"Microsoft.Authorization/roleAssignments/read\",
-\"Microsoft.Authorization/roleAssignments/delete\",
-\"Microsoft.Authorization/roleAssignments/write\",
-\"Microsoft.Authorization/roleDefinitions/read\",
-\"Microsoft.Authorization/roleDefinitions/write\",
-\"Microsoft.Authorization/roleDefinitions/delete\",
-\"Microsoft.Storage/storageAccounts/listkeys/action\",
-\"Microsoft.Storage/storageAccounts/delete\",
-\"Microsoft.Storage/storageAccounts/read\",
-\"Microsoft.Storage/storageAccounts/write\",
-\"Microsoft.Storage/storageAccounts/blobServices/containers/write\",
-\"Microsoft.Storage/storageAccounts/blobServices/containers/delete\",
-\"Microsoft.Storage/storageAccounts/blobServices/containers/read\",
-\"Microsoft.ManagedIdentity/userAssignedIdentities/delete\",
-\"Microsoft.ManagedIdentity/userAssignedIdentities/read\",
-\"Microsoft.ManagedIdentity/userAssignedIdentities/write\",
-\"Microsoft.ManagedIdentity/userAssignedIdentities/federatedIdentityCredentials/read\",
-\"Microsoft.ManagedIdentity/userAssignedIdentities/federatedIdentityCredentials/write\",
-\"Microsoft.ManagedIdentity/userAssignedIdentities/federatedIdentityCredentials/delete\",
-\"Microsoft.Storage/register/action\",
-\"Microsoft.ManagedIdentity/register/action\"
-"""
-    sts_role_name="${CLUSTER_NAME}-custom-role-sts"
-    sts_role_definition="${ARTIFACT_DIR}/azure-custom-role-definition-sts-minimal-permissions.json"
-    sts_sp_name="${CLUSTER_NAME}-sp-sts"
-    sts_sp_output=$(mktemp)
-
-    echo "Create SP with minimal permission for CCO"
-    create_role_definition_json "${sts_role_name}" "${sts_required_permissions}" "${sts_role_definition}"
-    create_custom_role "${sts_role_definition}" "${sts_role_name}"
-    # for destroy
-    custom_role_name_json=$(echo "${custom_role_name_json}" | jq -c -S ". +={\"ccoctl\":\"${sts_role_name}\"}")
-    echo "${custom_role_name_json}" > "${SHARED_DIR}/azure_custom_role_name"
-    sts_sp_role="${sts_role_name}"
-
-    create_sp_with_custom_role "${sts_sp_name}" "${sts_sp_role}" "${AZURE_AUTH_SUBSCRIPTOIN_ID}" "${sts_sp_output}"
-    sts_sp_id=$(jq -r .appId "${sts_sp_output}")
-    sts_sp_password=$(jq -r .password "${sts_sp_output}")
-    sts_sp_tenant=$(jq -r .tenant "${sts_sp_output}")
+sp_list=""
+[[ "${AZURE_SP_CREATION_FOR_CLUSTER}" == "true" ]] && sp_list="${sp_list} cluster"
+[[ "${ENABLE_MIN_PERMISSION_FOR_STS}" == "true" ]] && sp_list="${sp_list} sts"
+for sp_type in ${sp_list}; do
+    role_name=$(jq -r ".${sp_type}" "${SHARED_DIR}/azure_custom_role_name")
```

Comment:
In `cucushift-installer-rehearse-azure-ipi-private-networking-type-minimal-permission` case, there is no `azure_custom_role_name` generated, right?

Reply:
right, fixed, moved this line into line114 `if` block.